pkg/azurefile/azurefile.go

```diff
@@ -252,49 +252,38 @@ func GetFileShareInfo(id string) (string, string, string, string, error) {
 
 // check whether mountOptions contains file_mode, dir_mode, vers, if not, append default mode
 func appendDefaultMountOptions(mountOptions []string) []string {
-	fileModeFlag := false
-	dirModeFlag := false
-	versFlag := false
-	actimeoFlag := false
-	mfsymlinksFlag := false
-
-	for _, mountOption := range mountOptions {
-		if strings.HasPrefix(mountOption, fileMode) {
-			fileModeFlag = true
-		}
-		if strings.HasPrefix(mountOption, dirMode) {
-			dirModeFlag = true
-		}
-		if strings.HasPrefix(mountOption, vers) {
-			versFlag = true
-		}
-		if strings.HasPrefix(mountOption, actimeo) {
-			actimeoFlag = true
-		}
-		if strings.HasPrefix(mountOption, mfsymlinks) {
-			mfsymlinksFlag = true
-		}
+	var defaultMountOptions = map[string]string{
+		fileMode:   defaultFileMode,
+		dirMode:    defaultDirMode,
+		vers:       defaultVers,
+		actimeo:    defaultActimeo,
+		mfsymlinks: "",
 	}
 
-	allMountOptions := mountOptions
-	if !fileModeFlag {
-		allMountOptions = append(allMountOptions, fmt.Sprintf("%s=%s", fileMode, defaultFileMode))
-	}
+	// required for maintaing the order.
+	var keys = []string{fileMode, dirMode, vers, actimeo, mfsymlinks}
```

Comment:
why still need `keys`, keys is already in `defaultMountOptions`

Reply:
We need it to have a fixed iteration order. Map iteration is unordered and randomized. I thought we should keep a pre-defined order in the array. Otherwise, we can remove keys and then modify the unit tests.

Reply:
order is not important, feel free to modify the ut, thanks.

Reply:
Modified. Thanks.